app/src/main/java/org/simple/clinic/setup/SetupActivityUpdate.kt

```diff
@@ -17,22 +18,28 @@ class SetupActivityUpdate : Update<SetupActivityModel, SetupActivityEvent, Setup
       is UserDetailsFetched -> {
         val updatedModel = model
             .withLoggedInUser(event.loggedInUser)
-            .withSelectedCountry(None)
-        val effect = initialScreenEffect(event.loggedInUser, event.hasUserCompletedOnboarding)
+            .withSelectedCountry(event.userSelectedCountry)
+        val effect = initialScreenEffect(event.loggedInUser, event.hasUserCompletedOnboarding, event.userSelectedCountry)
 
         next(updatedModel, effect)
       }
       is DatabaseInitialized -> dispatch(FetchUserDetails)
+      is FallbackCountrySetAsSelected -> dispatch(GoToMainActivity)
       else -> noChange()
     }
   }
 
   private fun initialScreenEffect(
       loggedInUser: Optional<User>,
-      hasUserCompletedOnboarding: Boolean
+      hasUserCompletedOnboarding: Boolean,
+      selectedCountry: Optional<Country>
   ): SetupActivityEffect {
+    val hasUserLoggedInCompletely = loggedInUser is Just && selectedCountry is Just
```

Comment:
Do you also feel `.isNotEmpty()` has better readability, than comparing it to the type?

Reply:
Yes, it does. I'll make this change.